devenv/src/cnix.rs

```diff
@@ -144,19 +151,25 @@ impl Nix {
             .run_nix_with_substituters("nix", &args, &options)
             .await?;
 
+        // Delete any old generations of this profile.
         let options = Options {
             logging: false,
             ..self.options
         };
-
         let args: Vec<&str> = vec!["-p", gc_root_str, "--delete-generations", "old"];
         self.run_nix("nix-env", &args, &options).await?;
+
+        // Save the GC root for this profile.
         let now_ns = get_now_with_nanoseconds();
         let target = format!("{}-shell", now_ns);
-        symlink_force(
-            &fs::canonicalize(gc_root).expect("to resolve gc_root"),
-            &self.devenv_home_gc.join(target),
-        );
+        if let Ok(resolved_gc_root) = fs::canonicalize(gc_root) {
+            symlink_force(&resolved_gc_root, &self.devenv_home_gc.join(target));
+        } else {
+            warn!(
```

Comment:
Is this just a safety check in case `refresh_cached_output` doesn't take care of it?